content/en/docs/setup/production-environment/container-runtimes.md

```diff
@@ -151,10 +164,11 @@ Install containerd:
 {{< tabs name="tab-cri-containerd-installation" >}}
 {{% tab name="Linux" %}}
 
-1. Install the `containerd.io` package from the official Docker repositories. 
-Instructions for setting up the Docker repository for your respective Linux distribution and 
-installing the `containerd.io` package can be found at 
-[Install Docker Engine](https://docs.docker.com/engine/install/#server).
+1. Install the `containerd.io` package from the [official containerd website](
```

Comment:
There are no "containerd.io" packages on the containerd.io website, there it just called "containerd" (and no packages)

The alternative (to "containerd") would be "cri-containerd-cni" which also includes the CRI and CNI configuration:

`/etc/crictl.yaml`

`etc/cni/net.d/10-containerd-net.conflist`

Reply:
Where does the package name `cri-containerd-cni` come from?

Reply:
@sftim cri-containerd-cni is a tarball, recommended as a way to install by the containerd documentation: https://containerd.io/docs/getting-started/#starting-containerd

+1 for not mentioning containerd.io package.

Reply:
makes sense to link to https://containerd.io/downloads/ and https://containerd.io/docs/getting-started/#starting-containerd

Reply:
@afbjorklund installing CNI can cause [hard to debug issues](https://github.com/cri-o/cri-o/issues/2885). I'd recommend to install containerd from `cri-containerd` tarball instead.

Reply:
Kubernetes only works with one CNI at a time, installing podman or cri-o or even containerd is known to interfere with cluster setup...

The recommendation is to wipe /etc/cni before configuring and starting the cluster

Reply:
@afbjorklund  So, what's the point to install CNI if it's going to be removed anyway?

btw, I don't see any recommendation to remove CNI in [this documentation](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm/). Can you point me out to the recommendation you've mentioned?

Reply:
I don't think there is any instructions regarding CNI at all ? So it was mostly how I was adding myself.

`rm -f /etc/cni/net.d/*.conf*`

`kubectl apply -f https://raw.githubusercontent.com/flannel-io/flannel/v0.14.0/Documentation/kube-flannel.yml`

This of course also _breaks_ all other usage, so you need to move the podman configuration directory.

```
# Path to the directory where CNI configuration files are located.
#
# network_config_dir = "/etc/cni/net.d/"
```

Similar with the nerdctl configuration, so that it doesn't interfer with Kubernetes' containerd config (of CNI) ?

Reply:
@bart0sh I am not sure what the best way is to package this.

The user _needs_ to set up /etc/crictl.yaml and /etc/cni/net.d

But at the same time, there can be only one container runtime and only one container networking for Kubernetes.

So the files from one will conflict with the files from the other ones. Maybe they should go in /usr/share or in docs ?

Reply:
@afbjorklund installing https://github.com/containerd/containerd/releases/download/v1.6.0/cri-containerd-1.6.0-linux-arm64.tar.gz and then following [this guide](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm/) is the most straightforward way to get working kubernetes cluster in my opinion.

Reply:
Sure, but it doesn't have any details ? It says you need a CRI and you need a CNI, but after that it is just "GL;HF".

The entire process was more like: https://github.com/lima-vm/lima/blob/v0.8.3/examples/k8s.yaml#L28_L79

Reply:
[this guide](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm/) was referencing [container runtimes](https://kubernetes.io/docs/setup/production-environment/container-runtimes/) previously IIRC. So, I always followed first guide and it worked for me.

Reply:
Minor packaging detail: runc is bundled with "containerd.io" and with "cri-containerd", there is no "runc.io" package.

There is a regular system "runc" package, that is a dependency of the regular system "containerd" package though.

The docker packages are basically internal to docker, they could change (and have changed before) at any time.

Haven't looked at the rpm packaging (this was deb), but assume there is a similar vendor vs. system duplication ?

Reply:
For SIG Docs: we want to be as generic as we can be about third-party solutions. Even if someone uses containerd, we don't know what OCI runtime they prefer. `runc` is common but there are others that are no less valid.

Feedback on this PR should be aimed at helping the author, @PranshuSrivastava, get this ready for merge. For any other discussion, use issues or Slack.

Also:
```
$ dpkg-query --show runc
runc	1.0.1-0ubuntu2~20.04.1
```
(that's a `.deb`)


@PranshuSrivastava the feedback I'd highlight for you here is: https://github.com/kubernetes/website/pull/30882#discussion_r816258989